Neos.Neos/Classes/Domain/Service/FusionService.php

```diff
@@ -136,57 +137,40 @@ public function createRuntime(TraversableNodeInterface $currentSiteNode, Control
     public function getMergedFusionObjectTree(TraversableNodeInterface $startNode)
     {
         $siteResourcesPackageKey = $this->getSiteForSiteNode($startNode)->getSiteResourcesPackageKey();
-
         $siteRootFusionPathAndFilename = sprintf($this->siteRootFusionPattern, $siteResourcesPackageKey);
-        $siteRootFusionCode = $this->readExternalFusionFile($siteRootFusionPathAndFilename);
-
-        $mergedFusionCode = '';
-        $mergedFusionCode .= $this->generateNodeTypeDefinitions();
-        $mergedFusionCode .= $this->getFusionIncludes($this->prepareAutoIncludeFusion());
-        $mergedFusionCode .= $this->getFusionIncludes($this->prependFusionIncludes);
-        $mergedFusionCode .= $siteRootFusionCode;
-        $mergedFusionCode .= $this->getFusionIncludes($this->appendFusionIncludes);
 
-        return $this->fusionParser->parse($mergedFusionCode, $siteRootFusionPathAndFilename);
+        return $this->fusionParser->parseFrom(
+            FusionSourceCodeCollection::fromArray(array_filter([
+                ...$this->generateNodeTypeDefinitions(),
+                ...$this->prepareAutoIncludeFusion(),
+                ...$this->prepareFusionIncludes($this->prependFusionIncludes, $siteRootFusionPathAndFilename),
+                is_readable($siteRootFusionPathAndFilename)
+                    ? FusionSourceCode::fromFile($siteRootFusionPathAndFilename)
+                    : null,
+                ...$this->prepareFusionIncludes($this->appendFusionIncludes, $siteRootFusionPathAndFilename),
+            ]))
+        );
     }
```

Comment:
This array filtering and unpacking is not that easy to get how about 
```
new FusionSourceCodeCollection (
     $this->generateNodeTypeDefinitions(),
     $this->prepareAutoIncludeFusion(),
     $this->prepareFusionIncludes($this->prependFusionIncludes, $siteRootFusionPathAndFilename),
     is_readable($siteRootFusionPathAndFilename) ? FusionSourceCode::fromFile($siteRootFusionPathAndFilename) : FusionSourceCode::empty(),
     $this->prepareFusionIncludes($this->appendFusionIncludes, $siteRootFusionPathAndFilename),
);
```
this would require to change the constuctor of the FusionSourceCodeCollection to
```
class FusionSourceCodeCollection implements \IteratorAggregate, \Countable
{
    public function __construct(FusionSourceCode|FusionSourceCodeCollection ...$fusionCodeCollection)
```
and do the collection unpacking in there

also `FusionSourceCode::empty()` would remove the need for the filtering of null valies